tree/treeplayer/inc/ROOT/TAdoptAllocator.hxx

```diff
@@ -121,8 +125,18 @@ public:
          StdAllocTraits_t::deallocate(fStdAllocator, p, n);
    }
 
-   bool operator==(const TAdoptAllocator<T> &other) { return fAllocType == other.fAllocType; }
-   bool operator!=(const TAdoptAllocator<T> &other) { return fAllocType != other.fAllocType; }
+   bool operator==(const TAdoptAllocator<T> &other)
+   {
+      return fInitialAddress == other.fInitialAddress &&
+             fAllocType == other.fAllocType &&
+             fStdAllocator == other.fStdAllocator;
+   }
+   bool operator!=(const TAdoptAllocator<T> &other)
+   {
+      return fInitialAddress != other.fInitialAddress ||
+             fAllocType != other.fAllocType ||
+             fStdAllocator != other.fStdAllocator;
```

Comment:
`return !(*this == other)`?

Reply:
also.
For the usage of namespaces, this has been addressed. ADL is not yet there though.